schedule_lessons/dashboard/views.py

```diff
@@ -973,20 +992,47 @@ def error_check_and_save_lesson(request, lesson, context):
             context.get("no_ending_time_error") and not
             context.get("bigger_start_time_error") and not
             context.get("pending_relationship_error") and not
-            context.get("no_relationship_error") and not
-            context.get("")):
+            context.get("no_relationship_error")):
         start_time_in_local_time = datetime.datetime.combine(date, start_time,
                                                              time_difference)
         if start_time_in_local_time < datetime.datetime.now(tz=time_difference):
             context["past_lesson_error"] = (
                 "Fix starting time or date of lesson to make sure it's after "
                 "current time.")
             return context
-        context["status"] = 200
         end_time_in_local_time = datetime.datetime.combine(date, end_time,
                                                            time_difference)
         lesson.start_time = start_time_in_local_time.astimezone(UTC_ZONE)
         lesson.end_time = end_time_in_local_time.astimezone(UTC_ZONE)
+        availabilities = list(Availability.objects.filter(
+                                    profile__user=person_to_schedule_with,
+                                    day=lesson.start_time.strftime("%A")))
+        context['non_available_time_error'] = True
+        if availabilities:
```

Comment:
do we need this check? if availabilities is empty or null wont the for loop below just not execute?

Reply:
Yeah you're right